library/src/main/java/com/alamkanak/weekview/WeekView.java

```diff
@@ -1854,7 +2018,14 @@ private void goToNearestOrigin(){
 
         int nearestOrigin = (int) (mCurrentOrigin.x - leftDays * (mWidthPerDay + mColumnGap));
 
-        if (nearestOrigin != 0) {
+        if (mCurrentOrigin.x - nearestOrigin < getXMaxLimit()
+                && mCurrentOrigin.x - nearestOrigin > getXMinLimit()) {
```

Comment:
This boolean expression could be extracted to a variable with a useful name since it is used twice and makes the two `if`s long lines